builders/container-apps-internal-registry-demo/startup-script.sh

```diff
@@ -52,49 +52,124 @@ token=$(printf "%s" "$REGISTRY_AUTH_USERNAME:$REGISTRY_AUTH_PASSWORD" | base64)
 acr_access_string="Basic $token"
 export CNB_REGISTRY_AUTH='{"'$ACR_RESOURCE_NAME'":"'$acr_access_string'"}'
 
+echo "Initiating buildpack build..."
+echo "Correlation id: '$CORRELATION_ID'"
+echo 
+
+RETRY_DELAY=2
+RETRY_ATTEMPTS=5
+
+function fail_if_retry_exceeded() {
+  retries=$1
+  if [ "$retries" -ge $RETRY_ATTEMPTS ]; then
+    echo "----- Retry attempts exceeded -----"
+    exit 1
+  fi
+}
+
 # Execute the analyze phase
 echo
-echo "======================================="
 echo "===== Executing the analyze phase ====="
-echo "======================================="
-/lifecycle/analyzer \
-  -log-level debug \
-  -run-image mcr.microsoft.com/oryx/builder:stack-run-debian-bullseye-20230926.1 \
-  $APP_IMAGE
+retryCount=0
+until [ "$retryCount" -ge $RETRY_ATTEMPTS ]
+do
+  if [ "$retryCount" -ge 1 ]; then
+    echo "----- Retrying analyze phase (attempt $retryCount) -----"
+  fi
+
+  /lifecycle/analyzer \
```

Comment:
As of today, the CLI stops as soon as "ERROR:" is detected in the logs. However with the retry this wouldn't work anymore. I believe we might be able to have something where the CLI checks for "ERROR:" only if the logs are done streaming? If not, we might need to have some error interception here maybe. Nothing to change for this PR.

Reply:
so once all the retries complete, we exit with code 1. This actually will cause another build to be created (up to 4 times i think, provided time limit of 30 min has not been reached), so we get tons of retrying altogether for each build